index.js

```diff
@@ -351,8 +351,11 @@ class GithubScm extends Scm {
                   + '$CONFIG_URL $SD_CONFIG_DIR"; '
                   + 'else if [ -z $GIT_SHALLOW_CLONE_DEPTH ]; '
                   + 'then export GIT_SHALLOW_CLONE_DEPTH=50; fi; '
+                  + 'export GIT_SHALLOW_CLONE_BRANCH="--no-single-branch"; '
+                  + 'if [ $GIT_SHALLOW_CLONE_SINGLE_BRANCH = true ]; '
+                  + 'then export GIT_SHALLOW_CLONE_BRANCH="--single-branch"; fi; '
```

Comment:
just make this value empty for override.
ie, default behavior is DEPTH 50 & option `--no-single-branch`

user override should be GIT_SHALLOW_CLONE_SINGLE_BRANCH = true and that point, don't pass the option.

Reply:
Your current logic might be okay, I recall that some versions of git might not support this `--single-branch` as an option even though it is the default. But since we are bundling out own git, this should not be an issue. 

Just double check if git in macs has both options supported.

Reply:
Change to empty string for `--single-branch` so we don't need to worry about it